cli/offchain-commands/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+#[derive(Clone, Debug, clap::Subcommand)]
+#[allow(clippy::large_enum_variant)]
+pub enum OffchainCmd {
+	/// Get the script addresses for a given sidechain
+	Addresses,
+	/// Register a committee candidate
+	Register,
+	/// Deregister a committee candidate
+	Deregister,
+	/// Insert or update D parameter
+	UpsertDParameter,
+	/// Set or update permissioned candidates list
+	UpdatePermissionedCandidates,
+	/// Create a new token reserve
+	ReserveCreate,
+	/// Deposit assets to existing reserve
+	ReserveDeposit,
+	/// Empty and remove an existing reserve
+	ReserveHandover,
+	/// Release currently available funds from an existing
+	ReserveRelease,
+}
+
+#[derive(Clone, Debug, clap::Parser)]
+#[command(author, version, about, long_about = None)]
+pub struct CommonArguments<SidechainParams: clap::Args> {
+	#[clap(flatten)]
+	sidechain_params: SidechainParams,
```

Comment:
I've learned that the next smart-contracts release will have smart-contracts changed, so they do not take all params but only parameters relevant to their logic, besides genesis-utxo that would serve identifier role